resources/fileTemplates/internal/Magento UI Component Custom Data Provider Class.php.ft

```diff
@@ -102,10 +102,17 @@ class ${CLASS_NAME} extends ${EXTENDS}
             return $this->loadedData;
         }
         $this->loadedData = parent::getData();
+        $itemsById = [];
 
+#set($entityId = "#if(${ENTITY_ID})${ENTITY_ID}#{else}entity_id#end")
+#set($entityIdAccessor = '[(int) $item[' + "'$entityId'" + ']]')
+#set($getIdFromParamsAccessor = "[(int) $this->request->getParam('$entityId')]")
         foreach ($this->loadedData['items'] as $item) {
-        #set($idAccessor = '[' + "#if(${ENTITY_ID})'${ENTITY_ID}'#{else}'entity_id'#end" + ']')
-            $this->loadedData['items'][$item$idAccessor] = $item;
+            $itemsById$entityIdAccessor = $item;
```

Comment:
Doesn't seem a bit weird this sentence?

Reply:
@eduard13 it is not. We must do such things because the Apache Velocity template engine cannot parse $itemsById[(int) $item['entity_id']] or something similar.